### PR TITLE
Add rnix-lsp for Nix diagnostics and completion

### PR DESCRIPTION
**Pull request by jD91mZM2**

### CODE REVIEW

**Review by hsanson — CHANGES_REQUESTED** [review]
Please also include respective documentation changes and add some tests. Check these PRs as examples:

   - https://github.com/dense-analysis/ale/pull/1725/files
   - https://github.com/dense-analysis/ale/pull/1847/files

**Comment by jD91mZM2** [comment]
There, is that better? :)

File: test/nix/test.nix [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (hsanson) [inline test/nix/test.nix:1]: This file can be empty as we do not actually run anything in the test. Is just to check if the command to run it is correct.

**Review by hsanson — APPROVED** [review]

File: test/nix/test.nix (hunk 1 above)
Comment (jD91mZM2) [reply test/nix/test.nix]: Thanks, updated :)

**Review by hsanson — APPROVED** [review]
Good to go. Thanks or the work.

**Pull request merged**